primitives/derive/src/numeric.rs

```diff
@@ -0,0 +1,17 @@
+use core::str::FromStr;
+
+use manyhow::{error_message, Result};
+use proc_macro2::TokenStream;
+use quote::quote;
+
+pub fn numeric_impl(input: TokenStream) -> Result<TokenStream> {
+    let input = input.to_string();
+    let numeric = ::iroha_numeric::Numeric::from_str(&input)
+        .map_err(|err| error_message!("failed to parse numeric: {err}"))?;
+    let mantissa = numeric.mantissa();
+    let scale = numeric.scale();
+
+    Ok(quote! {
+        Numeric::new(#mantissa, #scale)
```

Comment:
```suggestion
        ::iroha_primitives::numeric::Numeric::new(#mantissa, #scale)
```

Reply:
I did that on purpose, since in the executor direct invocation of `iroha_primitives::numeric` is not possible since not in dependency list...